src/main/java/com/google/devtools/build/lib/bazel/repository/RepositoryResolvedEvent.java

```diff
@@ -114,78 +85,36 @@ public String getMessage() {
   }
 
   /** Returns an unstructured message explaining the origin of this rule. */
-  public static String getRuleDefinitionInformation(Rule rule) {
-    StringBuilder buf = new StringBuilder();
-
-    // Emit stack of rule instantiation.
-    buf.append("Repository ").append(rule.getName()).append(" instantiated at:\n");
-    ImmutableList<StarlarkThread.CallStackEntry> stack = rule.reconstructCallStack();
-    // TODO: Callstack should always be available for bazel.
-    if (stack.isEmpty()) {
-      buf.append("  callstack not available\n");
-    } else {
-      for (StarlarkThread.CallStackEntry frame : stack) {
-        buf.append("  ").append(frame.location).append(": in ").append(frame.name).append('\n');
-      }
-    }
-
-    // Emit stack of rule class declaration.
-    stack = rule.getRuleClassObject().getCallStack();
-    if (stack.isEmpty()) {
-      buf.append("Repository rule ").append(rule.getRuleClass()).append(" is built-in.\n");
-    } else {
-      buf.append("Repository rule ").append(rule.getRuleClass()).append(" defined at:\n");
-      for (StarlarkThread.CallStackEntry frame : stack) {
-        buf.append("  ").append(frame.location).append(": in ").append(frame.name).append('\n');
-      }
-    }
-
-    return buf.toString();
+  public static String getRuleDefinitionInformation(RepoDefinition repoDefinition) {
+    // We used to output a call stack for repos defined in WORKSPACE, but in Bzlmod we always get an
+    // empty stack -- for repos backing modules there's no call stack; for extension-generated
+    // repos, the call stack is lost during the roundtrip to the lockfile.
+    // TODO: store the call stack in the lockfile and output it here?
```

Comment:
If so, we should only store it in the hidden lockfile. I don't think it's worth the effort for now though.

Reply:
why so? Wouldn't that mean we don't have stack traces for any repos generated by non-reproducible extensions?

Reply:
The stack traces for extension-generated repos tend to be less useful since they usually point into the single generic instantiation point deep in the extension code. That's not where a user of the extension would typically want to look to address these warnings - they can't even directly update the repo rule attributes. 

I would say that we need to fundamentally rethink how "resolved attributes" should work with Bzlmod. As long as we have callstacks for `use_repo_rule`, the current state seems okay though. Could you file an issue to track this?

Reply:
Filed https://github.com/bazelbuild/bazel/issues/26511

Reply:
And added a TODO in RepositoryFetchFunction.